TradeTrust/tradetrust-website
# feat: upload csv to populate address book

# CSV file upload to populate Address Book

The goal of this PR is to expose a hook for the app to access the address book.

![image](https://user-images.githubusercontent.com/7406870/77145550-db61ad00-6ac3-11ea-91fb-aad734404bb5.png)

- Adds hook to read and update address book data in localstorage
- Adds button to read and process csv and call hook to update
- Adds raw address book view to view the storage until story on address book is done
- Temporarily smacked the button on header to access it before the story on address book is done, feature toggled off in prod

## Not Done
- Not added to storybook because cannot access localstorage from iframe
- Error handling in the visuals, currently errors are thrown directly to the alert, suggestion to pass in `onError` to the component so the parent component can show error message when there are errors
- Not doing any visual/e2e testing as this is suppose to be hidden away until the address book component is built

## To Test
- Download attached file and rename to .csv
- Drop any .tt file into the dropzone
- Click on "Upload Address Book" and select csv
- You should see the content of the address book
- Restart the session and see that the address book is still populated

[sampleaddress.txt](https://github.com/TradeTrust/tradetrust-website/files/4366929/sampleaddress.txt)
